src/bluesearch/entrypoint/database/convert_pdf.py

```diff
@@ -124,38 +145,124 @@ def run(
     int
         The exit code of the command
     """
-    # Check if the input file exists
-    if not input_pdf_path.exists():
-        logger.error(
-            f"The input file {str(input_pdf_path)!r} does not exist.",
-        )
+    # Check the GROBID server
+    if not grobid_is_alive(grobid_host, grobid_port):
+        logger.error("The GROBID server is not alive")
         return 1
 
-    # Check if the output file already exists
-    if output_xml_path.exists() and not force:
-        logger.error(
-            f"The output file {str(output_xml_path)!r} already exists. "
-            "Either delete it or use the --force option to overwrite it.",
-        )
+    # Check if the input file exists
+    if not input_path.exists():
+        logger.error(f"The input path {str(input_path)!r} does not exist")
         return 1
 
-    # Read the PDF file
-    logger.info("Reading the PDF file")
-    with input_pdf_path.open("rb") as fh_pdf:
-        pdf_content = fh_pdf.read()
+    # Collect input paths
+    input_paths: Iterable[pathlib.Path]
+    if input_path.is_file():
+        input_paths = [input_path]
+    else:
+        input_paths = input_path.rglob("*.pdf")
 
-    # Convert the PDF to XML
-    logger.info("Converting PDF to XML")
-    from bluesearch.database.pdf import grobid_pdf_to_tei_xml
+    failed_paths = []
+    path_map = _prepare_output_paths(input_paths, output_dir, force)
 
-    xml_content = grobid_pdf_to_tei_xml(pdf_content, grobid_host, grobid_port)
+    if len(path_map) == 0:
+        logger.warning("No files to process, stopping")
+        return 0
 
-    # Write the XML file
-    logger.info("Writing the XML file to disk")
-    with output_xml_path.open("w") as fh_xml:
-        n_bytes = fh_xml.write(xml_content)
-    logger.info("Wrote %d bytes to %s", n_bytes, output_xml_path.resolve().as_uri())
+    if output_dir is not None:
+        output_dir.mkdir(exist_ok=True)
+
+    # Convert
+    def do_work(pdf_path):
+        xml_path = path_map[pdf_path]
+        try:
+            _convert_pdf_file(grobid_host, grobid_port, pdf_path, xml_path)
+        except Exception as exc:
+            logger.exception(
+                f"An error happened when processing {pdf_path.resolve().as_uri()}: "
+                f"{exc}"
+            )
+            # lists seem to be thread-safe in Python
+            failed_paths.append(pdf_path)
 
-    logger.info("PDF conversion done")
+    with ThreadPoolExecutor(max_workers=num_workers) as executor:
+        executor.map(do_work, path_map)
```

Comment:
Adding a value for `timeout` ([doc](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.map)) will prevent the code to be stuck if the GROBID server is stuck.

Reply:
Good idea, I added a 60 second timeout in ac9d712, this should be generous enough for the PDF conversion.

Reply:
However, we're not handling the `TimeoutError` that will be thrown in this case, but maybe that's exactly what we want? If the GROBID server is stuck then it's probably better to just crash?

Reply:
Great!

Yes, I guess so.